File: all-submissions/0019-remove-nth-node-from-end-of-list/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        if (head == null) return null;
        if(head.next==null && n==1) return null;
        ListNode prev = null;
        ListNode slow = head;
        ListNode fast = head;
        int i=1;
        while(i <= n) {
            fast = fast.next;
            i++;
        }
        while(fast != null) {
            prev=slow;
            slow = slow.next;
            fast=fast.next;
        }
        if (slow == head) {
            head = head.next;
            return head;
        }
        prev.next = slow.next;
        return head;
    }
}

File: all-submissions/0019-remove-nth-node-from-end-of-list/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        k = head
        dummy = ListNode(0)
        dummy.next = head
        i = 1
        while(i != n and k is not None):
            k = k.next
            i +=1
        if (k is None): return head
        if (i==n and k.next is None):
            return  dummy.next.next
        curr = head
        prev = None
        while(k.next):
            k = k.next
            prev = curr
            curr = curr.next
        prev.next = curr.next
        return dummy.next

